nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/util/SecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,191 @@
+                        cmdLogger.error("Encountered a security exception while generating secure Apache NiFi configuration.", e);
+                        throw new RuntimeException(e);
+                    }
+
+                    // Move over the new stores from temp dir
+                    Files.move(new File(tlsConfiguration.getKeystorePath()).toPath(), new File(keystorePath).toPath(),
+                            StandardCopyOption.REPLACE_EXISTING);
+                    Files.move(new File(tlsConfiguration.getTruststorePath()).toPath(), new File(truststorePath).toPath(),
+                            StandardCopyOption.REPLACE_EXISTING);
+
+                    updateProperties(propertiesFile, tlsConfiguration);
+
+                    cmdLogger.info("Successfully generated {} and {}.", keystorePath, truststorePath);
+                } else if (!keystore.exists() && truststore.exists()) {
+                    cmdLogger.error("Tried to generate keystore {} for secure Apache NiFi configuration, but truststore file {} already exists.  Aborting.",
+                            keystorePath, truststorePath);
+                    throw new RuntimeException("Will not generate keystore and truststore separately.");
+                } else if (keystore.exists() && !truststore.exists()) {
+                    cmdLogger.error("Tried to generate truststore {} for secure Apache NiFi configuration, but keystore file {} already exists.  Aborting.",
+                            truststorePath, keystorePath);
+                    throw new RuntimeException("Will not generate keystore and truststore separately.");
+                } else {
+                    cmdLogger.info("Existing keystore and truststore detected: skipping Apache Nifi certificate generation.");
+                }
+            } else {
+                cmdLogger.warn("HTTPS is configured, but keystore and truststore are not specified.  This will result in an invalid configuration.");
+            }
+        } else {
+            cmdLogger.info("No HTTPS configuration detected: skipping Apache Nifi certificate generation.");
+        }
+    }
+
+    /**
+     * Attempts to add some reasonable guesses at desired SAN values that can be added to the generated
+     * certificate.
+     * @param nifiProperties The nifi.properties
+     * @return A Pair with IP SANs on the left and DNS SANs on the right
+     */
+    private static Pair<String[], String[]> getSubjectAlternativeNames(Properties nifiProperties) {
+        Set<String> ipSubjectAlternativeNames = new HashSet<>();
+        Set<String> dnsSubjectAlternativeNames = new HashSet<>();
+
+        try {
+            ipSubjectAlternativeNames.add(InetAddress.getLocalHost().getHostAddress());
+            dnsSubjectAlternativeNames.add(InetAddress.getLocalHost().getHostName());
+        } catch (UnknownHostException e) {
+            // This was just a courtesy, so we'll skip adding it
```

Comment:
In contrast to some other comments, adding a debug log message here seems more useful than the comment with an empty catch block.